js-tests/src/shortcodeModelSpec.js

```diff
@@ -58,16 +64,19 @@ describe( "Shortcode Model", function() {
 		var _shortcode = jQuery.extend( true, {}, shortcode );
 
 		// Test with attribute and with content.
-		expect( _shortcode.formatShortcode() ).toEqual( '[test_shortcode attr="test value"]test content[/test_shortcode]' );
+		expect( _shortcode.formatShortcode() ).toEqual( '[test_shortcode attr="test value" checkbox="true"]test content[/test_shortcode]' );
 
 		// Test without content.
 		_shortcode.get('inner_content').unset( 'value' );
-		expect( _shortcode.formatShortcode() ).toEqual( '[test_shortcode attr="test value"]' );
+		expect( _shortcode.formatShortcode() ).toEqual( '[test_shortcode attr="test value" checkbox="true"]' );
 
 		// Test without attributes
 		_shortcode.get( 'attrs' ).first().unset( 'value' );
-		expect( _shortcode.formatShortcode() ).toEqual( '[test_shortcode]' );
+		expect( _shortcode.formatShortcode() ).toEqual( '[test_shortcode checkbox="true"]' );
 
+		// Test with falsey checkbox
+		_shortcode.get( 'attrs' ).last().set( 'value', 'false' );
+		expect( _shortcode.formatShortcode() ).toEqual( '[test_shortcode checkbox="false"]' );
```

Comment:
@danielbachhuber That would evaluate as null, which would trigger the default to override (Which is the behavior we are trying to prevent).